sympy/physics/continuum_mechanics/truss.py

```diff
@@ -0,0 +1,620 @@
+            self._loads.pop(location)
+
+    def apply_support(self, location, type):
+        """
+        This method adds a pinned or roller support at a particular node
+
+        Parameters
+        ==========
+
+        location: String or Symbol
+            Label of the Node at which support is added.
+
+        type: String
+            Type of the support being provided at the node.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.add_node('B', 3, 0)
+        >>> t.apply_support('A', 'pinned')
+        >>> t.supports
+        {'A': 'pinned'}
+        """
+        if location not in self._node_labels:
+            raise ValueError("Support must be added on a known node")
+
+        else:
+            if location not in list(self._supports):
+                if type == 'pinned':
+                    self.apply_load(location, Symbol('R_'+str(location)+'_x'), 0)
+                    self.apply_load(location, Symbol('R_'+str(location)+'_y'), 90)
+                elif type == 'roller':
+                    self.apply_load(location, Symbol('R_'+str(location)+'_y'), 90)
+            elif self._supports[location] == 'pinned':
+                if type == 'roller':
+                    self.remove_load(location, Symbol('R_'+str(location)+'_x'), 0)
+            elif self._supports[location] == 'roller':
+                if type == 'pinned':
+                    self.apply_load(location, Symbol('R_'+str(location)+'_x'), 0)
+            self._supports[location] = type
+
+    def remove_support(self, location):
+        """
+        This method removes support from a particular node
+
+        Parameters
+        ==========
+
+        location: String or Symbol
+            Label of the Node at which support is to be removed.
+
+        Examples
+        ========
+
+        >>> from sympy.physics.continuum_mechanics.truss import Truss
+        >>> t = Truss()
+        >>> t.add_node('A', 0, 0)
+        >>> t.add_node('B', 3, 0)
+        >>> t.apply_support('A', 'pinned')
+        >>> t.supports
+        {'A': 'pinned'}
+        >>> t.remove_support('A')
+        >>> t.supports
+        {}
+        """
+        if location not in self._node_labels:
+            raise ValueError("No such node exists in the Truss")
+
+        elif location not in list(self._supports):
+            raise ValueError("No support has been added to the given node")
+
+        else:
+            if self._supports[location] == 'pinned':
+                self.remove_load(location, Symbol('R_'+str(location)+'_x'), 0)
+                for load in self._loads[location]:
+                    if load[1] == 90:
+                        load[0] -= Symbol('R_'+str(location)+'_y')
+                        if load[0] == 0:
+                            self._loads[location].remove(load)
+                        break
```

Comment:
You should treat both x and y reactions equally, you are calling remove load for x but not for y.
This will give an error if a load is applied in x and then support is applied and removed. For example:
```
>>> from sympy.physics.continuum_mechanics.truss import Truss
>>> t = Truss() 
>>> t.add_node("node_1", 0, 0)
>>> t.apply_load("node_1",10,0)
>>> t.apply_support("node_1","pinned")
>>> t.remove_support("node_1")
```
I think after you update `apply_load` according to my comment above, you will just have to call `remove_load` for both x and y and it will work.